docs/pull_requests.md

```diff
@@ -58,3 +58,17 @@ later identify and filter out the intermediate commits if desired, e.g. with
 We've disabled the "Rebase and merge" option, since it does a fast-forward
 merge, which makes the intermediate commits indistingishuable from the validated
 and reviewed final state of the PR.
+
+A common task is to rebase a PR on main so that it is up to date, and push
+the updated branch to test it in the CI before merging. Suppose a user
+`contributor` submitted a branch `bugfix` as PR `1234`, and has allowed
+the maintainers to update the PR. Then you could quickly perform a rebase:
```

Comment:
I think this is worth adding either way, but is there a reason to do this over the ["Update branch > Update with rebase"](https://github.blog/changelog/2022-02-03-more-ways-to-keep-your-pull-request-branch-up-to-date/) button? (I've never tried using the button on someone else's PR, so I don't know what it does to authorship information.)

Reply:
I'm not sure either, maybe that button could work in the simple case! (I didn't notice that it existed, or else I subconsciously ignored it.) I guess the documented method is useful for more complex work, like yeah squashing or if you have to rebase and then fix stuff or append to a PR.

Reply:
I rewrote as

```markdown
A common task is to rebase a PR on main so that it is up to date, perhaps fix
some conflicts or add some changes to the PR, and then push the updated branch
to test it in the CI before merging. Suppose a user `contributor` submitted a
branch `bugfix` as PR `1234`, and has allowed the maintainers to update the PR.
Then you could fetch the branch to perform work on the PR locally:

    git fetch origin pull/1234/head:pr-1234
    git checkout pr-1234
    git rebase main
    <fix conflicts or commit other changes>
    git push -f git@github.com:contributor/shadow.git pr-1234:bugfix
    git checkout main
    git branch -D pr-1234

If it passes the tests, you can merge the PR in the Github interface as usual.
```

But the PR auto-merged in the meantime so I guess I won't  bother updating again. :)

Reply:
I plan on making a documentation PR, so I'll include this in that PR.